### PR TITLE
feat(core): implemented platform merchant account

### PR DESCRIPTION
**Pull request by racnan**
## Type of Change

- [ ] Bugfix
- [X] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

#### About the feature
- This the first PR for the Platform Account feature.
- Feature is under development and behind the feature flag. 
- Feature is supposed to be disabled for sandbox and production environments.
#### Enabling the platform account.
- New API has been created to enable platform account.
- API requires Admin API auth.
- Once enabled, there is no way of disabling platform account as of now.
#### How it works
- A new special header is to be sent by the platform called `x-connected-merchant-id`.
- Platform will use its own API key irrespective of the merchant id present in the above header.
- Value of this header is supposed to be the merchant id of the merchant account on behalf of which the platform is performing the operation.
- **If the platform has `is_platform` true and the merchant account on behalf of which operation is performed belongs to the same organisation as platform** then the operation is allowed.
#### Payment Intent
- Operation will be performed for merchant id present in the `x-connected-merchant-id` header.
- `platform_merchant_id` column for the payment intent will be populated with platforms merchant id.
#### Other operations
- Will add context of `platform_merchant_id` in other operations as well.

### Additional Changes

- [x] This PR modifies the API contract
- [x] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context
#6883 

## How did you test it?
1. Create a merchant account(m1).
2. Use below api to convert it into platform account.
```sh
curl --location --request POST '<BASE URL>/accounts/{merchant_id}/platform' \
--header 'api-key:<ADMIN API KEY>'
```
3. Create another merchant account in the same organisation(m2).
4. Do a payment using platform's(m1) API key for connected merchant account(m2) by passing header `x-connected-merchant-id` with value as connected merchant account's merchant id.
5. Check if payment is created for connected merchant account.
6. Check if `platform_merchant_id` is populated with platform's merchant id.

## Checklist

- [X] I formatted the code `cargo +nightly fmt --all`
- [X] I addressed lints thrown by `cargo clippy`
- [X] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

**Review by jarnura — DISMISSED** [review]

File: crates/router/src/core/payments/operations/payment_confirm.rs [hunk 1]
```diff
@@ -77,6 +77,7 @@ impl<F: Send + Clone + Sync> GetTracker<F, PaymentData<F>, api::PaymentsRequest>
         key_store: &domain::MerchantKeyStore,
         auth_flow: services::AuthFlow,
         header_payload: &hyperswitch_domain_models::payments::HeaderPayload,
+        _platform_merchant_account: Option<&domain::MerchantAccount>,
```
Comment (Narayanbhat166) [inline crates/router/src/core/payments/operations/payment_confirm.rs:80]: we should validate that this belongs to the same merchant_id that was passed in create right?

File: crates/router/src/services/authentication.rs [hunk 2]
```diff
@@ -548,8 +573,31 @@ where
             .await
             .to_not_found_response(errors::ApiErrorResponse::Unauthorized)?;
 
+        // Get connected merchant account if API call is done by Platform merchant account on behalf of connected merchant account
+        let (merchant, platform_merchant_account) = if state.conf().platform.enabled {
+            get_platform_merchant_account(state, request_headers, merchant).await?
```
Comment (Narayanbhat166) [inline crates/router/src/services/authentication.rs:578]: we can pass key store from this function itself

**Review by Narayanbhat166 — DISMISSED** [review]

File: crates/router/src/core/payments/operations/payment_confirm.rs (hunk 1 above)
Comment (racnan) [reply crates/router/src/core/payments/operations/payment_confirm.rs:80]: yess, will add validations in eventual PRs.

**Review by pixincreate — APPROVED** [review]
configs and ci, look good to me

**Review by NishantJoshi00 — APPROVED** [review]
The `HeaderAuth` side of things look good

**Review by Narayanbhat166 — DISMISSED** [review]

File: config/deployments/integration_test.toml
Comment (SanchithHegde) [inline config/deployments/integration_test.toml:1]: Can you make the same changes in the other config files as well: `config/development.toml`, `config/docker_compose.toml` and `config/deployments/env_specific.toml`?

File: crates/router/src/services/authentication.rs [hunk 3]
```diff
@@ -724,12 +775,14 @@ where
 }
 
 #[cfg(all(feature = "partial-auth", feature = "v1"))]
+#[allow(unused)]
```
Comment (SanchithHegde) [inline crates/router/src/services/authentication.rs:778]: Do we still need this annotation?

File: crates/router/src/services/authentication.rs [hunk 4]
```diff
@@ -1412,15 +1537,20 @@ pub struct MerchantIdAuth(pub id_type::MerchantId);
 
 #[cfg(feature = "v1")]
 #[async_trait]
+#[allow(unused)]
```
Comment (SanchithHegde) [inline crates/router/src/services/authentication.rs:1540]: Same here.

**Review by Narayanbhat166 — APPROVED** [review]

**Review by SanchithHegde — DISMISSED** [review]

**Review by Narayanbhat166 — APPROVED** [review]

**Review by SanchithHegde — APPROVED** [review]

**Review by jarnura — APPROVED** [review]

**Review by apoorvdixit88 — APPROVED** [review]

**Pull request merged**